src/Sequence.php

```diff
@@ -99,6 +99,27 @@ public function apply(callable $callback)
     }
 
     /**
+     * @param mixed $value
+     * @param null|int $num
+     */
+    public function fill($value, int $num = null)
+    {
+        if (null === $num) {
+            \array_walk($this->array, function (&$v) use ($value) {
+                $v = $value;
+            });
+        } else {
+            if ([] === $this->array) {
+                $this->array = \array_fill(0, $num, $value);
+            } else {
+                for ($i = 0; $i < $num; $i++) {
+                    $this->array[$i] = $value;
+                }
+            }
```

Comment:
For simplicity (and the fact that performance isn't critical for the polyfill), I think we can skip the empty array check and only run the loop.